src/evict/evict_lru.c

```diff
@@ -1012,7 +1019,7 @@ retry:	while (slot < max_entries && ret == 0) {
 				cache->evict_file_next = NULL;
 			else
 				dhandle = TAILQ_FIRST(&conn->dhqh);
-		} else {
+		} else if (!F_ISSET(dhandle, WT_DHANDLE_DEAD) || dead_next) {
```

Comment:
Is it possible that focusing on a single file will starve out evictions from other files, specifically meaning eviction doesn't see pages marked as ``WT_READGEN_OLDEST`` for longer?